### DISCUSSION TITLE
Sankey - Designating specific node colors

### DISCUSSION THREAD

**Discussion by dupchurch-sedaro** [plouc/nivo#1828] [Q&A]
Hi. First of all thanks for the amazing package. I had a question about specifying colors for specific nodes when defining the node data of a Sankey diagram. I noticed in the example from https://nivo.rocks/sankey/ that the data itself has a color field but then it seems to be overwritten by the themed color. Is there a way to turn off the themed color and instead just use that specified color on the node itself? 

The sankey diagram I am trying to create essentially always has 6 columns of nodes and I'm trying to have all nodes in a single column have the same color but I'm not sure how to do that with theme. An extension of this would be having static flow colors. So for example having the connections be a static grey while assigning the nodes colors when the data is defined.

Sorry if I missed the solution in the docs and thank you in advanced for any assistance.

**Comment by dupchurch-sedaro** [comment]
In case anyone else stumbles upon this I was able to get it to work but adding the `startColor` and `endColor` fields to the links object and keeping `enableLinkGradient` as true. Then I just set whatever colors I wanted there.

**Reply by PulkitBanta** [reply]
This only changes the color for the links and not for the node. Were you able to define a color for the node as well? If yes can you create a sample code snippet?

**Reply by mayank-permiso** [reply]
@dupchurch-sedaro , any more props you found which are not mentioned in Nivo documentation but can be given in json? BTW, thanks for the start and end color props, this will help me in changing the link color for each link, as there is no prop that can change each link color. 
Also, how did you find these 2 props? Any d3 documentation you read?

**Comment by 8Almd5bx** [comment]
Hi, I am a physician resident and need a sankey diagram four one of our project. Is there anyone who can provide this sankey diagram professionally? I'll be very happy to contact them separately and will greatly appreciate if you contact me. Thank you!

Cansu

**Comment by abdulhaseeb13mar** [comment]
i checked their types and found out that there is no color field for data (but their docs show there is one). no wonder why it was not working for node color

**Comment by dorin-moldoveanu-assist** [comment]
Hey there,

Any updates on this issue ? I would also really like to set my own **node** specific colors.
Thanks.

**Comment by plouc** [comment]
The property is `colors`, and it exists in the types as well.

**Comment by mayank-permiso** [comment]
This worked for me- 
` colors={(rest)=>  rest.nodeColor}`

Where nodeColors is defined for each node in data json. Yes, it accepts a function. This should have been mentioned in the docs, but I think it is not.
